### ISSUE TITLE
[Question] Using TB's SwaggerUI to create a GraphQL Schema. Is the Swagger specs compliant?

### ISSUE THREAD

**Issue by GrayCygnus** [thingsboard/thingsboard#9009]
**Component**
 * Generic

**Description**
TL; DR: How can I generate a GraphQL schema "automagically" with tools like https://github.com/yarax/swagger-to-graphql or https://github.com/IBM/openapi-to-graphql ? Tried them but they complain on TB's SwaggerUI/OpenAPI specs and format (incomplete? non-compliant?). Any guidance greatly appreciated :)

---

Longer version:

Hey everybody, hope things are great. I've been itching to try GraphQL with ThingsBoard's API to level up our deployment and tech stack. To achieve so a GraphQL Schema has to be created; basically a file that tells GraphQL about the objects, endpoints and API calls available, so GraphQL can then "use them accordingly" to query the data.

As TB's API is not small, writing it by hand seems like a daunting task. Also, tools like the two gits shared above exist, suggesting that a Schema can be "automagically" created based on a SwaggerUI/OpenAPI specification. As TB has SwaggerUI (kudos on that) I tried the tools on it, but was unable to obtain such Schema. 

I raised an Issue on Jira, but after some discussion they suggested I posted this here on Github, as (parraphrasing) "they weren't too familiar with GraphQL and SwaggerUI is intended as documentation and hasn't been tested with tools like these or with GraphQL".

Any guidance, or tip or support is greatly appreciated. I would like to avoid having to manually create the Schema, as I'm also no expert on GraphQL, TB's API is not small, etc..

Here's the output I get when trying one of those tools:

![GraphQLSwaggerUITB-git](https://github.com/thingsboard/thingsboard/assets/7274943/9b815aee-a031-4410-bbd9-e676ff16ad49)

I'll try checking with the folks over those tools to get their guidance as well and let you all know what I find.

Thanks,
Mike

---

**Environment**
 * OS:  any
 * ThingsBoard: any

**Comment by mykhailokornieiev** [comment]
As the community has not provided assistance on this matter over an extended period, I am closing this ticket due to inactivity. If it is still relevant, feel free to open a new ticket with a full description of your request.
